android/src/main/java/com/mr/flutter/plugin/filepicker/FileUtils.java

```diff
@@ -314,14 +315,25 @@ public static FileInfo openFileStream(final Context context, final Uri uri, bool
                     fos.getFD().sync();
                 }
             } catch (final Exception e) {
-                try {
-                    fos.close();
-                } catch (final IOException | NullPointerException ex) {
-                    Log.e(TAG, "Failed to close file streams: " + e.getMessage(), null);
-                    return null;
-                }
                 Log.e(TAG, "Failed to retrieve path: " + e.getMessage(), null);
                 return null;
+            } finally {
+                if (fos != null) {
+                    try {
+                        fos.close();
```

Comment:
If this call fails, the closing of the input stream is never tried? Can we do the `return null;` at the very end of the finally block? Or will that not have any meaningful effect?

Reply:
Sounds like a valid concern, I pushed a fix.